grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1449,7 +1449,7 @@ class ProductService {
 
     Product addSynonymToProduct(String productId, String synonymTypeCodeName, String synonymValue, String localeName) {
         Product product = Product.get(productId)
-        Locale locale = localeName ? new Locale(localeName) : null
+        Locale locale = localeName ? LocalizationUtil.getLocale(localeName) : null
```

Comment:
Not a question to be resolved here, but I'm wondering if there's a fancier way to do this so that we don't need to bake all this localization logic into our services.

It'd be awesome if we could mark a Domain field as "localized" and then whenever we do a look up on that field it'd automatically also search for the synonyms.

Reply:
Yeah, I guess this is a more Synonyms related topic suiting for a TD discovery or some OBIG